velox/expression/CastExpr.cpp

```diff
@@ -380,14 +399,28 @@ VectorPtr CastExpr::applyArray(
   }
   context.swapErrors(oldErrors);
 
-  // Assemble the output array
+  // Returned array vector should be addressable for every element, even those
+  // that are not selected.
+  BufferPtr sizes = input->sizes();
+  if (newElements->isConstantEncoding()) {
+    // If the newElements we extends its size since that is cheap.
+    newElements->resize(input->elements()->size());
+  } else if (newElements->size() < input->elements()->size()) {
```

Comment:
We need to reset the sizes and offsets for non-selected rows even if newElements->size() == input->elements()->size(), right? Because these non-selected rows may have offset larger than the max offsets of selected rows.

Reply:
if the new elements vectors size is >= the old one, that mean every position in non selected rows can still be referenced although we do not care about that.

Reply:
Discussed with Laith offline. I'm good with this code.